### PR TITLE
Update Gradle plugins and dependencies

### PR DESCRIPTION
**Pull request by arturdryomov**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -22,5 +22,4 @@ android {
 dependencies {
     implementation project(':kronos-android')
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    testImplementation 'junit:junit:4.12'
```
Comment (arturdryomov) [inline app/build.gradle:25]: Unused.

File: kronos-java/build.gradle [hunk 2]
```diff
@@ -5,8 +5,9 @@ apply from: "$rootDir/gradle/publish-java.gradle"
 dependencies {
     api "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
 
-    testImplementation 'junit:junit:4.12'
     testImplementation "com.nhaarman:mockito-kotlin:1.6.0"
-    testImplementation 'org.mockito:mockito-core:2.19.0'
-    testImplementation 'org.assertj:assertj-core:3.9.0'
+    testImplementation "junit:junit:4.13.1"
+    testImplementation "org.mockito:mockito-core:3.6.0"
+    testImplementation "org.mockito:mockito-inline:3.6.0"
```
Comment (arturdryomov) [inline kronos-java/build.gradle:11]: This artifact includes the deleted resource file below.

File: kronos-java/src/test/java/com/lyft/kronos/internal/ntp/SntpServiceTest.kt [hunk 3]
```diff
@@ -69,8 +69,10 @@ class SntpServiceTest {
         whenever(sntpClient.requestTime("1.us.pool.ntp.org", TIMEOUT_MS)).thenReturn(mockResponse)
         whenever(sntpClient.requestTime("2.us.pool.ntp.org", TIMEOUT_MS)).thenReturn(mockResponse)
         whenever(mockResponse.currentTimeMs).thenReturn(-1)
+
         assertThat(sntpService.sync()).isFalse()
 
-        verify(sntpSyncListener, times(1)).onError(any(), any<IllegalStateException>())
+        verify(sntpSyncListener, times(1)).onError(eq("1.us.pool.ntp.org"), any<NTPSyncException>())
+        verify(sntpSyncListener, times(1)).onError(eq("2.us.pool.ntp.org"), any<NTPSyncException>())
```
Comment (arturdryomov) [inline kronos-java/src/test/java/com/lyft/kronos/internal/ntp/SntpServiceTest.kt:76]: This is a bit weird. Updating Kotlin results in this:

```
org.mockito.exceptions.verification.TooManyActualInvocations: 
syncListener.onError(
    <any java.lang.String>,
    <any com.lyft.kronos.internal.ntp.NTPSyncException>
);
Wanted 1 time:
-> at com.lyft.kronos.internal.ntp.SntpServiceTest.throw error if response has negative time value(SntpServiceTest.kt:74)
But was 2 times:
-> at com.lyft.kronos.internal.ntp.SntpServiceImpl.sync(SntpService.kt:160)
-> at com.lyft.kronos.internal.ntp.SntpServiceImpl.sync(SntpService.kt:160)
```

Looked through the code and seems like `sync` goes through all servers, calling listener for each server. So the adjustment I’ve made makes sense but not sure why the Kotlin version resulted in a behavior change. Might be bytecode changes and how Mockito manipulates it.

**Comment by arturdryomov** [comment]
PTAL @ameliariely, @buildbreaker

**Review by buildbreaker — APPROVED** [review]

**Pull request merged**